src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,930 @@
+            Next
+            Return S.ToString
+        End Function
+
+
+        '''**************************************************************************
+        ''' RemoveEndingSeparator
+        ''' <summary>
+        ''' Removes all directory separators at the end of a path.
+        ''' </summary>
+        ''' <param name="Path">a full or relative path.</param>
+        ''' <returns>If Path is a root path, the same value. Otherwise, removes any directory separators at the end.</returns>
+        ''' <remarks>We decided not to return path with separators at the end.</remarks>
+        Private Shared Function RemoveEndingSeparator(ByVal Path As String) As String
+            If IO.Path.IsPathRooted(Path) Then
+                ' If the path is rooted, attempt to check if it is a root path.
+                ' Note: IO.Path.GetPathRoot: C: -> C:, C:\ -> C:\, \\myshare\mydir -> \\myshare\mydir
+                ' BUT \\myshare\mydir\ -> \\myshare\mydir!!! This function will remove the ending separator of
+                ' \\myshare\mydir\ as well. Do not use IsRoot here.
+                If Path.Equals(IO.Path.GetPathRoot(Path), StringComparison.OrdinalIgnoreCase) Then
+                    Return Path
+                End If
+            End If
+
+            ' Otherwise, remove all separators at the end.
+            Return Path.TrimEnd(IO.Path.DirectorySeparatorChar, IO.Path.AltDirectorySeparatorChar)
+        End Function
+
+        Private Function FileHasExpectedDate(FileNameWithPath As String, ExpectedData() As Char) As Boolean
+            Using stream As New IO.StreamReader(IO.File.OpenRead(FileNameWithPath))
+                Dim ReadData(ExpectedData.Length - 1) As Char
+                stream.Read(ReadData, 0, SourceData.Length)
+                Return ExpectedData = ReadData
+            End Using
+        End Function
+        <Fact>
+        Public Shared Sub CombinePathTest_BadBaseDirectory_RelativePath()
+            Assert.Throws(Of ArgumentNullException)(Function() FileSystem.CombinePath("", "Test2"))
+        End Sub
+
+        <Fact>
+        Public Shared Sub CombinePathTest_BaseDirectory_RelativePath()
+            Using TestBase As New FileIOTests
+                Dim TestDirInfo As New IO.DirectoryInfo(TestBase.TestDirectory)
+                Dim Root As String = TestDirInfo.Root.Name
+                Assert.Equal(FileSystem.CombinePath(Root, "Test2"), IO.Path.Combine(Root, "Test2"))
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub CombinePathTest_RootDirectory_RelativePath()
+            Using TestBase As New FileIOTests
+                Assert.Equal(FileSystem.CombinePath(TestBase.TestDirectory, ""), TestBase.TestDirectory)
+                Assert.Equal(FileSystem.CombinePath(TestBase.TestDirectory, "Test"), IO.Path.Combine(TestBase.TestDirectory, "Test"))
+            End Using
+        End Sub
+        <Fact>
+        Public Shared Sub IOPathCombineTest()
```

Comment:
>IOPathCombineTest [](start = 26, length = 17)

Consider changing name to be consistent with earlier tests and to describe test. Perhaps `CombinePathTest_Nothing`.

And please test `Nothing` for `relativePath` as well.